server/controllers/chapterItemsController.js

```diff
@@ -1,10 +1,31 @@
 const AppError = require('../utils/AppError');
+const express = require('express');
+const db = require ('../models.js');
 
 chapterItemsController = {};
 
 
 chapterItemsController.addItem = async (req, res, next) => {
-  return next(new AppError(new Error('Not implemented'), 'chapterItemsController', 'addItem', 500));
+  const {
+    item_id,
+    total_needed,
+    total_received
+  } = req.body;
+
+  const {chapter_id} = req.params;
+
+  //query that updates the totals based on the items table-- make separate controller for this and put in middleware
+
+  // const total_needed= *****;
+ 
+  try {
+    const text = 'INSERT INTO public.chapter_items (chapter_id, item_id, total_needed, total_received)VALUES ($1, $2, $3, $4) RETURNING * ';
+    const params = [chapter_id, item_id, total_needed, total_received];
+    const res = await db.query(text, params);
+    console.log(`Successfully added ${res.rows[0]} to the database`);
+  } catch (err) {
+    return next(new AppError(new Error('Not implemented'), 'chapterItemsController', 'addItem', 500));
```

Comment:
This error message should be more descriptive. I would provide the caught err object as the first argument to AppError

Reply:
That makes sense. So you mean passing in err instead of 'not implemented'?

Reply:
Yeah, exactly just `new AppError(err, 'controller_name', 'method_name', status)`